File: Edu-Kotlin/src/com/jetbrains/edu/kotlin/twitter/KtOptionsProvider.java
package com.jetbrains.edu.kotlin.twitter;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.project.ProjectManager;
import com.intellij.ui.components.JBCheckBox;
import com.jetbrains.edu.learning.StudyTaskManager;
import com.jetbrains.edu.learning.TwitterPluginConfigurator;
import com.jetbrains.edu.learning.StudyUtils;
import com.jetbrains.edu.learning.settings.StudyOptionsProvider;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;


public class KtOptionsProvider implements StudyOptionsProvider {
  private KtTwitterSettings twitterSettings;
  private JBCheckBox myAskToTweetCheckBox;
  private JPanel myPanel;
  private boolean myIsModified = false;

  KtOptionsProvider() {
    Project[] openProjects = ProjectManager.getInstance().getOpenProjects();
    for (Project project : openProjects) {
      if (StudyTaskManager.getInstance(project).getCourse() != null) {
        TwitterPluginConfigurator twitterConfigurator = StudyUtils.getTwitterConfigurator(project);
        if (twitterConfigurator != null) {
          twitterSettings = KtTwitterSettings.getInstance(project);
          myAskToTweetCheckBox.setSelected(twitterSettings.askToTweet());
          break;
        }
      }
    }
    myAskToTweetCheckBox.addActionListener(e -> myIsModified = true);
  }

  @Override
  public void apply() {
    twitterSettings.setAskToTweet(myAskToTweetCheckBox.isSelected());
  }

  @Override
  public void reset() {
    twitterSettings.setAskToTweet(true);
  }

  @Override
  public void disposeUIResources() {

  }

  @Nullable
  @Override
  public JComponent createComponent() {
    return myPanel;
  }

  @Override
  public boolean isModified() {
    return myIsModified;
  }
}

File: Edu-Kotlin/src/com/jetbrains/edu/kotlin/twitter/KtOptionsProvider.kt
package com.jetbrains.edu.kotlin.twitter

import com.intellij.openapi.project.Project
import com.intellij.ui.IdeBorderFactory
import com.intellij.ui.components.JBCheckBox
import com.intellij.ui.layout.*
import com.jetbrains.edu.kotlin.messages.EduKotlinBundle
import com.jetbrains.edu.learning.EduUtils
import com.jetbrains.edu.learning.settings.OptionsProvider
import org.jetbrains.annotations.Nls
import javax.swing.JComponent

class KtOptionsProvider(private val project: Project) : OptionsProvider {

  private val askToTweetCheckBox: JBCheckBox = JBCheckBox(
    EduKotlinBundle.message("ktoptions.ask.to.tweet"),
    KtTwitterSettings.getInstance(project).askToTweet()
  )

  @Nls
  override fun getDisplayName(): String = EduKotlinBundle.message("twitter.settings")

  override fun isModified(): Boolean {
    return KtTwitterSettings.getInstance(project).askToTweet() != askToTweetCheckBox.isSelected
  }

  override fun apply() {
    KtTwitterSettings.getInstance(project).setAskToTweet(askToTweetCheckBox.isSelected)
  }

  override fun reset() {
    askToTweetCheckBox.isSelected = KtTwitterSettings.getInstance(project).askToTweet()
  }

  override fun createComponent(): JComponent? {
    // Default project can be passed here while building searchable options
    if (project.isDefault) return null
    if (!EduUtils.isEduProject(project)) return null
    if (EduUtils.getTwitterConfigurator(project) == null) return null

    return panel {
      row { askToTweetCheckBox() }
    }.apply {
      border = IdeBorderFactory.createTitledBorder("Kotlin")
    }
  }
}
